nservicebus/best-practices.md

```diff
@@ -0,0 +1,130 @@
+
+An NServiceBus endpoint is designed to be a long-lived object that persists for the entire life of the application process. Once the `IMessageSession` is created, use dependency injection to inject it into controllers or wherever else it is needed. If necessary, assign the `IMessageSession` to a global variable.
+
+
+:x: **DO NOT use messaging for [data distribution](/nservicebus/messaging/data-distribution.md)**
+
+A common example of a data distribution scenario is having cached data on multiple scaled-out web servers and attempting to deliver a message to each of them. The message indicates each server should drop their current cache entries and retrieve fresh data from the database.
+
+Asynchronous messaging (e.g. NServiceBus) is **not** a good solution for data distribution scenarios. Other technologies that are designed specifically to solve these problems should be used instead. Some examples can be found in the [recommendations section](/nservicebus/messaging/data-distribution.md#recommendations) of the data distribution article.
+
+
+## Message endpoints
+
+:heavy_check_mark: **DO keep message handlers focused on business code**
+
+Message handlers should be fairly simple, and focused on only the business code needed to handle the message. Infrastructure code for logging, exception management, timing, auditing, authorization, unit of work, message signing/encryption (the list could go on forever) should not be included in a message handler.
+
+Instead, implement this functionality separately in a [message pipeline behavior](/nservicebus/pipeline/manipulate-with-behaviors.md), which enables inserting additional functionality into the NServiceBus message processing pipeline, kind of like an [ASP.NET ActionFilter except for messaging.
+
+For a high-level overview of infrastructure concerns and behaviors, see the blog post [Infrastructure soup](https://particular.net/blog/infrastructure-soup).
+
+:x: **DO NOT catch exceptions in a message handler**
+
+NServiceBus relies on a message handler completing _without_ throwing an exception to know that the message was processed successfully. If an exception is thrown, the message can be retried until it succeeds, which resolves many transient errors. (See the blog post [I caught an exception. Now what?](https://particular.net/blog/but-all-my-errors-are-severe).) If too many retries occur, the message is moved to an error queue where a developer can figure out what's wrong with it and correct the underlying issue.
+
+If exceptions are caught in a message handler, the message will be consumed, without the ability to be retried, regardless of the error.
```

Comment:
This doesn't read well?
- What does it mean 'caught'? Maybe mention "swallowed" as they basically disappear?
- What does it mean messages are consumed? Maybe mention "removed from the queue" or something?